### PR TITLE
S4 course page

### LINKED ISSUE #107: Course page
As a user, I want to be able to view a page which summarizes course information and shows reviews and questions for that course, and gives me the option to submit reviews and questions.

Acceptance tests:
- Course page with header showing course name and space for future aggregated rating/question information
- 'Tabs' for Info, Questions and Reviews
- For Question and Review tabs, a 'feed' or list of items, which can be clicked to go to the single-page version of the question or review
- For info tab, a space for the most important course information; i.e. scraped Handbook description etc

### PR DESCRIPTION
**Pull request by NunoDasNeves**
resolves #107

### CODE REVIEW

File: frontend/src/App.vue [hunk 1]
```diff
@@ -34,17 +34,22 @@ html, body {
   --border-thick: 2px solid rgba(160, 178, 178, 0.5);
   --box-shadow-active: 0px 0px 0px 1px #ddd;
   --theme: #00a99d;
-  --theme-light: rgba(#00a99d, 0.6);
+  /*--theme-light: rgba(#00a99d, 0.6);*/
+  --theme-light: rgba(102, 203, 196);
```
Comment (DarkPurple141) [inline frontend/src/App.vue]: rgba with no a?

File: frontend/src/App.vue [hunk 2]
```diff
@@ -34,17 +34,22 @@ html, body {
...
   --font-small: 16px;
   --font-small-mobile: 12px;
 }
 
-main {
+#app {
+  background-color: var(--color-light-gray);
+}
+
+.main-content {
```
Comment (DarkPurple141) [inline frontend/src/App.vue]: not sure what I think of this XD.

Comment (DarkPurple141) [reply frontend/src/App.vue]: Maybe it makes sense I guess. Only, main is already meant to be main content.

File: frontend/src/App.vue [hunk 3]
```diff
@@ -34,17 +34,22 @@ html, body {
...
+  /*--theme-light: rgba(#00a99d, 0.6);*/
+  --theme-light: rgba(102, 203, 196);
   --font-large: 40px;
   --font-large-mobile: 24px;
   --font-small: 16px;
   --font-small-mobile: 12px;
 }
 
-main {
+#app {
```
Comment (DarkPurple141) [inline frontend/src/App.vue]: If you did * background-color: white and #app background-color: grey wouldn't it be more comprehensive rather the below subrules. Basically everything except for main should be white right?

File: frontend/src/main.js [hunk 4]
```diff
@@ -8,6 +8,7 @@ import LoadingSpinner from '@/components/LoadingSpinner'
 import AppLogo from '@/components/AppLogo'
 
 Vue.config.productionTip = false
+Vue.config.devtools = true
```
Comment (DarkPurple141) [inline frontend/src/main.js]: ?

File: frontend/src/utils/api/course.js [hunk 5]
```diff
@@ -0,0 +1,31 @@
+import { get } from './index'
```
Comment (DarkPurple141) [inline frontend/src/utils/api/course.js]: u can just write as from '.'

File: frontend/src/App.vue (hunk 2 above)
Comment (NunoDasNeves) [reply frontend/src/App.vue]: alternative is to put a main tag in each view

File: frontend/src/App.vue (hunk 1 above)
Comment (NunoDasNeves) [reply frontend/src/App.vue]: yep

File: frontend/src/App.vue (hunk 3 above)
Comment (NunoDasNeves) [reply frontend/src/App.vue]: no cos you need spaces between stuff to show the background. See course page

File: frontend/src/main.js (hunk 4 above)
Comment (NunoDasNeves) [reply frontend/src/main.js]: Vue browser extension doesn't work for production builds unless this is on

File: frontend/src/utils/api/course.js (hunk 5 above)
Comment (NunoDasNeves) [reply frontend/src/utils/api/course.js]: ok, but this is copied from your code in questions.js

**Comment by NunoDasNeves** [comment]
@DarkPurple141 i'm building each time yah

File: frontend/src/App.vue (hunk 1 above)
Comment (DarkPurple141) [reply frontend/src/App.vue]: fair

File: frontend/src/App.vue (hunk 2 above)
Comment (DarkPurple141) [reply frontend/src/App.vue]: Purpose of the main tag is to be the view container though, where header and footer are agnostic of view.

File: frontend/src/App.vue (hunk 3 above)
Comment (DarkPurple141) [reply frontend/src/App.vue]: Not sure I fully understand what you mean.

Comment (DarkPurple141) [reply frontend/src/App.vue]: The spaces come from margins and everything that isn't covering the fallback main/app will be grey?

File: frontend/src/utils/api/course.js (hunk 5 above)
Comment (DarkPurple141) [reply frontend/src/utils/api/course.js]: I make mistakes too 👍

File: frontend/src/components/questions-answers/QuestionCard.vue [hunk 6]
```diff
@@ -3,17 +3,17 @@
...
             <div class="meta-fields">
                 <p class="arrow">&#8679;</p>
-                <p class="likes">{{ question.likes }}</p>
+                <p class="likes">{{ likes }}</p>
                 <p class="arrow">&#8681;</p>
             </div>
             <div class="content">
-                <router-link tag="h2" :to="{ name: 'question', query: { qid: String(question.id) }}">
-                    {{ question.title }}
+                <router-link tag="h2" :to="{ name: 'question', query: { qid: String(id) }}">
```
Comment (DarkPurple141) [inline frontend/src/components/questions-answers/QuestionCard.vue]: change back to param for router-link here. query change was just to allow new questions.

File: frontend/src/components/questions-answers/QuestionCard.vue [hunk 7]
```diff
@@ -25,11 +25,15 @@ import Card from '@/components/Card'
 export default {
   components: { Card },
   props: {
-    question: Object
+    id: Number,
+    author: Number,
+    likes: Number,
+    title: String,
+    body: String,
+    published: String
```
Comment (DarkPurple141) [inline frontend/src/components/questions-answers/QuestionCard.vue]: Just to explain this.. when the component is actually rendered the prop has already been transformed to a string in the store. Thus, not a Date obj

File: frontend/src/main.js (hunk 4 above)
Comment (DarkPurple141) [reply frontend/src/main.js]: I do npm run serve and npm start in each folder so it's always on dev branch.

File: frontend/src/store/course/index.js [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+    name: '',
+    rating: 0
+  },
+  error: {
+    code: 0,
+    message: ''
+  }
+}
+
+const getters = {
```
Comment (DarkPurple141) [inline frontend/src/store/course/index.js]: If you're not doing anything special with the getters and you're using mappers in the component you don't really need getters.

File: frontend/src/utils/api/questions.js [hunk 9]
```diff
@@ -1,13 +1,6 @@
 import { get, post, put } from './index'
```
Comment (DarkPurple141) [inline frontend/src/utils/api/questions.js]: the offending line XD

File: frontend/tests/unit/Feed.spec.js [hunk 10]
```diff
@@ -1,13 +1,51 @@
...
-      propsData: { title }
+import { mount, RouterLinkStub } from '@vue/test-utils'
+import Feed from '@/components/course/Feed'
+import ReviewCard from '@/components/reviews-replies/ReviewCard'
+import QuestionCard from '@/components/questions-answers/QuestionCard'
+
+describe('Feed.vue Reviews', () => {
+  before(function() {
+    this.items = [
+      { id: 1, title: 'hey', body: 'wow', likes: 2, published: new Date() },
```
Comment (DarkPurple141) [inline frontend/tests/unit/Feed.spec.js]: Technically these props are wrong now for date. Doesn't seem to break things tho.

File: frontend/src/components/questions-answers/QuestionCard.vue (hunk 6 above)
Comment (NunoDasNeves) [reply frontend/src/components/questions-answers/QuestionCard.vue]: yes i've fixed this now

File: frontend/src/components/questions-answers/QuestionCard.vue (hunk 7 above)
Comment (NunoDasNeves) [reply frontend/src/components/questions-answers/QuestionCard.vue]: fair, fixed this in tests now :)

File: frontend/src/store/course/index.js (hunk 8 above)
Comment (NunoDasNeves) [reply frontend/src/store/course/index.js]: true but ceebs to change it now lel

File: frontend/tests/unit/Feed.spec.js (hunk 10 above)
Comment (NunoDasNeves) [reply frontend/tests/unit/Feed.spec.js]: i've fixed all the tests now

**Comment by NunoDasNeves** [comment]
I've fixed a few things, updated the nav, fixed tests and added a questionform test

**Comment by NunoDasNeves** [comment]
@Aruzal @PapaPwn probably a good idea for one of you to take a look at this before I merge

File: frontend/src/views/Course.vue [hunk 11]
```diff
@@ -1,18 +1,57 @@
...
+        </div>
+        <Feed
+          feedType="QuestionCard"
+          :items="questions"
+        />
+      </div>
+
+      <div v-if="courseTab=='reviews'">
+        <div class='button-container'>
+            <router-link :to="{ name: 'review', query: {cid: courseInfo.id} }">
```
Comment (Aruzal) [inline frontend/src/views/Course.vue]: Should be newReview like above.

**Pull request merged**